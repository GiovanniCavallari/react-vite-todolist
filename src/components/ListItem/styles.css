.list-item-container {
  display: flex;
  align-items: center;
  justify-content: space-between;
  flex-direction: row;
  border: 1px solid #ccc;
  padding: 8px;
  border-radius: 4px;
}

.list-item-container:not(:first-child) {
  margin-top: -1px;
}

.list-item-text {
  font-size: 14px;
}

.done {
  text-decoration: line-through;
}

.list-item-button-container button:first-child {
  margin-right: 8px;
}
