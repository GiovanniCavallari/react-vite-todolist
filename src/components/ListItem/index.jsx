import React from 'react';
import P from 'prop-types';
import Button from '../Button';
import { FaCheck, FaTrashAlt } from 'react-icons/fa';

import './styles.css';

function ListItem({ text, done, onFinish, onRemove }) {
  return (
    <div className="list-item-container" data-testid="list-item">
      <p className={`list-item-text ${done && 'done'}`}>{text}</p>

      <div className="list-item-button-container">
        <Button type="success" onClick={onFinish}>
          <FaCheck />
        </Button>
        <Button type="danger" onClick={onRemove}>
          <FaTrashAlt />
        </Button>
      </div>
    </div>
  );
}

ListItem.propTypes = {
  text: P.string.isRequired,
  done: P.bool.isRequired,
  onFinish: P.func.isRequired,
  onRemove: P.func.isRequired,
};

export default ListItem;
