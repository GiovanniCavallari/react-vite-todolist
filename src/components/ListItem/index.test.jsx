import { fireEvent, render } from '@testing-library/react';
import ListItem from './index';

describe('Test ListItem Component', () => {
  const callback = jest.fn();
  const text = 'sample to do';

  const component = ({ done }) => (
    <ListItem text={text} done={done} onFinish={callback} onRemove={callback} />
  );

  afterAll(() => {
    jest.clearAllMocks();
  });

  it('should render correctly', () => {
    const { asFragment } = render(component({ done: false }));
    expect(asFragment).toMatchSnapshot();
  });

  it('should item be done', () => {
    const { asFragment, getByText } = render(component({ done: true }));
    const p = getByText(text);

    expect(asFragment).toMatchSnapshot();
    expect(p.className).toContain('done');
  });

  it('should click on callbacks', () => {
    const { asFragment, getAllByTestId } = render(component({ done: false }));
    const buttons = getAllByTestId('button');

    buttons.forEach((btn) => {
      fireEvent.click(btn);
    });

    expect(asFragment).toMatchSnapshot();
    expect(callback).toBeCalledTimes(2);
  });
});
