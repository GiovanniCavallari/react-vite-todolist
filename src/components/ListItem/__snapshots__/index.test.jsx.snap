// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ListItem Component should click on callbacks 1`] = `[Function]`;

exports[`Test ListItem Component should item be done 1`] = `[Function]`;

exports[`Test ListItem Component should render correctly 1`] = `[Function]`;
